### PR TITLE
Issue 29 log folder bug

### LINKED ISSUE #29: [BUG] - log folder created locally, although specified outside of cwd
**Describe the bug**
The title describes the bug.

**Expected behavior**
If I ask ashpy to use a logdir outside of the cwd, nothing in the cwd should change.
Instead, currently, the JSON of the best model is created in the local directory.

**Code to reproduce the issue**
```
def get_model():
    """Create a new autoencoder tf.keras.Model."""
    autoencoder = Autoencoder(
        (64, 64),
        (4, 4),
        kernel_size=3,
        initial_filters=16,
        filters_cap=64,
        encoding_dimension=50,
        channels=3,
    )

    # encoding, representation = autoencoder(input)
    inputs = tf.keras.layers.Input(shape=(64, 64, 3))
    _, reconstruction = autoencoder(inputs)
    model = tf.keras.Model(inputs=inputs, outputs=reconstruction)
    return model


def _train(dataset: tf.data.Dataset, logdir: Path):
    reconstruction_error = ClassifierLoss(tf.keras.losses.MeanSquaredError())
    autoencoder = get_model()

    ClassifierTrainer(
        model=autoencoder,
        optimizer=tf.optimizers.Adam(1e-4),
        loss=reconstruction_error,
        metrics=[ashpy.metrics.ClassifierLoss(model_selection_operator=operator.lt)],
        logdir=str(logdir),
        epochs=100,
    )(dataset, dataset)
```

### PR DESCRIPTION
**Pull request by mr-ubik**
# Solve #29 

## Description

Moved the automatic log folder creation present in the Metrics, now it is triggered ONLY during the initialization of the various Trainers.
If a Trainer does something fancy with the metrics after the `super().__init__` call now it must manually (re)trigger `super().update_metrics(metrics)` Seet the two GANs Trainers as an example.

**NOTE:** with this change, Trainers always override the `logdir` argument passed to a Metric during its initialization. This was done already but each metric also self updated polluting the logsdir leading to thi bug.

Fixes #29 (issue)

## Type of change

- [x] Bugfix (i.e., a non-breaking change which fixes an issue)
- [x] Breaking change (i.e., a fix or feature that would cause existing functionality to not work as expected)
- [?] This change requires a documentation update

## How Has This Been Tested?

Complete test suite with `tox`

### Dependencies

Add `sphinx-copybutton` (it's awesome).

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [?] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by galeone — APPROVED** [review]
Looks great.
